watertap/tools/parameter_sweep/parameter_sweep_writer.py

```diff
@@ -222,27 +183,48 @@ def _write_outputs(self, output_dict, txt_options="metadata"):
         # We will also create a companion txt file by default which contains
         # the metadata of the h5 file in a user readable format.
         txt_fname = self.config["h5_results_file_name"] + ".txt"
-        if txt_options == "metadata":
-            my_dict = copy.deepcopy(output_dict)
-            for key, value in my_dict.items():
-                for subkey, subvalue in value.items():
-                    subvalue.pop("value")
-        elif txt_options == "keys":
-            my_dict = {}
-            for key, value in output_dict.items():
-                if key != "solve_successful":
-                    my_dict[key] = list(value.keys())
-        else:
-            my_dict = output_dict
 
-        with open(txt_fname, "w") as log_file:
-            pprint.pprint(my_dict, log_file)
+        if self.config.h5_parent_group_name is not None and os.path.isfile(txt_fname):
+            return
+        else:
+            if txt_options == "metadata":
+                my_dict = copy.deepcopy(output_dict)
+                for key, value in my_dict.items():
+                    for subkey, subvalue in value.items():
+                        subvalue.pop("value")
+            elif txt_options == "keys":
+                my_dict = {}
+                for key, value in output_dict.items():
+                    if key != "solve_successful":
+                        my_dict[key] = list(value.keys())
+            else:
+                my_dict = output_dict
+
+            with open(txt_fname, "w") as log_file:
+                pprint.pprint(my_dict, log_file)
 
     def _write_output_to_h5(self, output_dict, h5_results_file_name):
 
-        f = h5py.File(h5_results_file_name, "w")
+        if self.config.h5_parent_group_name is None:
+            # No parent groups exists, a new file will be created regardless
+            f = h5py.File(h5_results_file_name, "w")
+            parent_grp = f
+        elif (
+            len(self.config.h5_parent_group_name) > 0
```

Comment:
Is this condition necessary (it's not covered through exclusion)? What happens if `self.config.h5_parent_group_name == ""`?

Reply:
Ah! that was a remnant from a previous implementation. Removed in the new one.